src/components/events.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { StaticQuery, graphql } from 'gatsby';
+import { contentExists } from '../utils/ug-utils.js';
+import moment from 'moment';
+import '../styles/events.css';
+
+function findMatches(fromDrupal, fromWP) {
+    for (let i=0; i < fromDrupal.length; i++) {
+        for (let j=0; j < fromWP.length; j++) {
+            return fromDrupal[i].name === fromWP[j].name;
+        }
+    }
+}
+
+const generateEvents = (data, eventData) => {
+    let drupalCategories = eventData.relationships.field_event_category;
+    let matchAllCategories = eventData.field_match_categories;
+    let title = eventData.field_title;
+    let allEvents = data.allWpEvent.edges;
+    let shownEvents = [];
+    
+    if (matchAllCategories) {
+        for (let i=0; i < allEvents.length; i++) {        
+            if (JSON.stringify(drupalCategories) === JSON.stringify(allEvents[i].node.eventsCategories.nodes)) {
+                shownEvents.push(allEvents[i]);
+            }        
+        } 
+    } else {
+        for (let i=0; i < allEvents.length; i++) {
+            if (findMatches(drupalCategories, allEvents[i].node.eventsCategories.nodes)) {
+                shownEvents.push(allEvents[i]);
+            }
+        } 
+    }
+    
+    return (contentExists(shownEvents) ?
+    <React.Fragment key={eventData.drupal_id}>
+        <h2 className="mb-5">{contentExists(title) ? title : "Upcoming Events"}</h2>
+        <div className="event-list row gx-3 gy-5 justify-content-md-center mb-5">

```

Comment:
Is there a reason we're center-justifying the content instead of having it left-aligned?
<img width="1313" alt="image" src="https://user-images.githubusercontent.com/1927902/155193837-69bb8e88-ea1b-4297-8763-cf221549a549.png">

Also there is a bit of visual oddness when shrinking the viewport. Are we able to fix that?
<img width="1009" alt="image" src="https://user-images.githubusercontent.com/1927902/155194346-1815515d-69b0-47c7-a34d-61c72a44375b.png">

Reply:
YES! It looks fantastic.